frontend/public/kinds.ts

```diff
@@ -35,8 +34,8 @@ export const connectToPlural: ConnectToPlural = connect((state: State, props: {p
   const plural = props.plural || _.get(props, 'match.params.plural');
 
   const kindObj = isGroupVersionKind(plural)
-    ? state.k8s.getIn(['RESOURCES', 'models']).get(plural)
-    : _.find(k8sModels, model => model.plural === plural);
+    ? state.k8s.getIn(['RESOURCES', 'models', plural])
+    : state.k8s.getIn(['RESOURCES', 'models']).find(model => model.plural === plural);
```

Comment:
This allows the models to be defined in the plugin rather than in common k8sModels module

Reply:
I think this will break some other cases. For instance, if there are multiple `deployments` in different API versions, we might get the wrong one. We need to make sure we get the one defined in the static model.

Reply:
I agree that Console should keep referring to existing model definitions (`public/models/index.ts`) while **also** allowing plugins to contribute new model definitions.

I think the best way to solve this is via auto-generated `@console/plugin-models` module which exports additional model definitions for Console application to work with.

@jelkosz ^^ please create "Allow plugins to contribute new model definitions" task & assign it to me.

Reply:
@vojtechszocs ok, done

Reply:
> I think the best way to solve this is via auto-generated `@console/plugin-models` module which exports additional model definitions for Console application to work with.

I've re-thinked this and will avoid generating another virtual module in favor of collecting models from extension objects at runtime and augmenting `public/models/index.ts` module exports.

Reply:
#1586 should fix this.